crates/crates_io_trustpub/src/access_token.rs

```diff
@@ -0,0 +1,110 @@
+use secrecy::{ExposeSecret, SecretString};
+use sha2::digest::Output;
+use sha2::{Digest, Sha256};
+
+/// A temporary access token used to publish crates to crates.io using
+/// the "Trusted Publishing" feature.
+#[derive(Debug)]
+pub struct AccessToken(SecretString);
+
+impl AccessToken {
+    const PREFIX: &str = "cio_tp_";
+
+    /// Generate a new access token.
+    pub fn generate() -> Self {
+        Self::from_u64s(rand::random(), rand::random())
+    }
+
+    /// Create an access token from two u64 values.
+    ///
+    /// This is used internally by the `generate()` fn and is extracted
+    /// to a separate function for testing purposes.
+    fn from_u64s(r1: u64, r2: u64) -> Self {
+        let plaintext = format!("{}{r1:016x}{r2:016x}", Self::PREFIX);
+        Self(SecretString::from(plaintext))
+    }
```

Comment:
- is 128bits of randomness enough?
- https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program#identify-your-secrets-and-create-regular-expressions suggests adding a (32bit) checksum at the end for validation purposes. That would require us to find and add yet another dependency. Not sure if that's really worth it? 🤔

Reply:
I've completely rebuilt the `AccessToken` struct now.

It now uses 31 random alphanumeric characters and another alphanumeric character as a checksum, plus the prefix mentioned above. 

This is obviously not a cryptographic hash/checksum, but for the purposes that we want to use it for (preventing database interaction if the checksums don't match for a user-provided token) this shouldn't matter (unless I'm missing something 😅).

Another review on this part of the PR would be appreciated :)